### ISSUE TITLE
CKV_SECRETS_3: Azure storage account access key triggered for GCP KMS encrypted secrets

### ISSUE THREAD

**Issue by Daniel-I-Am** [bridgecrewio/checkov#6002]
**Describe the issue**
The CKV_SECRETS_3 check lints for Azure keys. My Terraform repository does not have any Azure keys, but does have a lot of committed [GCP KMS encrypted secrets](https://cloud.google.com/security/products/security-key-managements). The check is triggered for quite a few KMS secrets, not all.

I would assume the check is currently too permissive in what counts as an Azure storage account key.

**Examples**
For this example, I encrypted some dummy data using `gcloud kms encrypt`.

```
$ cat main.tf
variable "var_one" {
  type  = string
  value = "CiQAi6tN/XVboTYZMGPiaMMl0Ox5dhq17W10IPzyoejdM7vLUoUSPwCIKuTuA4V1FgfWnWXNcNRV1ptbh2YQdplQ6voHyz6gCAH2OTk4goQbIOH+I3JS+OPLFbzxOanfrD3ZlAqYWQ=="
}
variable "var_two" {
  type  = string
  value = "CiQAi6tN/Q0+wrrsNxUdee12qFr8ohWeiRs3Any32UHTyJDJ+EsSPQCIKuTuFlyW5UcdiW57V6LEkazuViRFGzR1dicfaNxdoDvCPRWbh3jza0Sr0KtZ2+NMDia0FsW9SvpOIyA="
}
$ docker run --rm -it -v "$(pwd):/tf" -w "/tf" bridgecrew/checkov --directory . --compact --quiet
secrets scan results:

Passed checks: 0, Failed checks: 1, Skipped checks: 0

Check: CKV_SECRET_3: "Azure Storage Account access key"
	FAILED for resource: 4acc496969ee53fdc6a7b22b529241b55f623721
	File: /main.tf:3-4
	Guide: https://docs.prismacloud.io/en/enterprise-edition/policy-reference/secrets-policies/secrets-policy-index/git-secrets-3
```

I would expect this not to trigger a secret check, as it is encrypted data.

**Version (please complete the following information):**
- Docker container, running
```
docker run --rm -it bridgecrew/checkov --version
3.2.8
```

**Comment by rensvanleeuwen** [comment]
Seeing the same with AWS CDK on generated CloudFormation. Example:

```json
{
  "CDKMetadata": {
    "Type": "AWS::CDK::Metadata",
    "Properties": {
      "Analytics": "v2:deflate64:H4sIAAAAAAAA/21SwU7DMAz9Fu5ZGEMIceymVf13knZdEXDy83svju1kxu/uHvn0RtR2IrNiUqMSbQVHVoBy4oTb2StY2nxFk1qUN1sSsmCe+myKyvLmFc5ssdchRKUSNO7iBYtdWiq5dakGCtyIEnQE7yItYeRHLriQdaFmJuTe8Yb4y6I5BGHDLQM54szvKwIWpRKkUF/NASzXcQgbQStBUIszi406eTgWXmsC4/Fg6Du5ZBH5Ob8q0MSeS6x940Ef4AaoRlNEslxqMt3R31RXfS8k+CGXOjui8qW2IJ1RdL4u5A/hOxjcLVOi4k2C/eRdjNFvqbuvRy2z97yZO1n0M11QH0bzz7xlqctyIMs7IUBmwtoefKX+GV7QEmQfqLuLx6xtWQIWnZFecJawGlPf4P9SbPCkMjBzYYH55wLyXyhXOm+Zxgz4wd6eZlP+5D/awSo1MU6TqoAnffwG7Qn1HoUCAAA="
    },
    "Metadata": {
      "aws:cdk:path": "OfdMainBootstrapStack/CDKMetadata/Default"
    }
  }
}
```

It breaks on the (value of) the `Analytics` tag it seems.

**Comment by Daniel-I-Am** [comment]
I did some digging into the changes in the past week. This check seems to be actually implemented in the [bc-detect-secrets](https://github.com/bridgecrewio/detect-secrets) project. Recently, a change was added (released in 1.5.3 of the be-detect-secrets project) that affected the Azure storage keys filter:
- https://github.com/bridgecrewio/detect-secrets/pull/178

The affected check seems to trigger for any string that is 86-1000 characters long and ends with `==`, which is most base64 encoded strings.

---

The updated package was included in [checkov 3.2.8](https://github.com/bridgecrewio/checkov/releases/tag/3.2.8), which seems to cause the issue. I tested with the sample above on both checkov version 3.2.8 and 3.2.7 and this is where the erroneous result is introduced.

```
$ cat main.tf
variable "var_one" {
  type  = string
  value = "CiQAi6tN/XVboTYZMGPiaMMl0Ox5dhq17W10IPzyoejdM7vLUoUSPwCIKuTuA4V1FgfWnWXNcNRV1ptbh2YQdplQ6voHyz6gCAH2OTk4goQbIOH+I3JS+OPLFbzxOanfrD3ZlAqYWQ=="
}
variable "var_two" {
  type  = string
  value = "CiQAi6tN/Q0+wrrsNxUdee12qFr8ohWeiRs3Any32UHTyJDJ+EsSPQCIKuTuFlyW5UcdiW57V6LEkazuViRFGzR1dicfaNxdoDvCPRWbh3jza0Sr0KtZ2+NMDia0FsW9SvpOIyA="
}
$ docker run --rm -it -v "$(pwd):/tf" -w "/tf" bridgecrew/checkov:3.2.7 --directory . --compact --quiet

$ docker run --rm -it -v "$(pwd):/tf" -w "/tf" bridgecrew/checkov:3.2.8 --directory . --compact --quiet
secrets scan results:

Passed checks: 0, Failed checks: 1, Skipped checks: 0

Check: CKV_SECRET_3: "Azure Storage Account access key"
	FAILED for resource: 4acc496969ee53fdc6a7b22b529241b55f623721
	File: /main.tf:3-4
	Guide: https://docs.prismacloud.io/en/enterprise-edition/policy-reference/secrets-policies/secrets-policy-index/git-secrets-3
```

---

I hope the additional information helps the resolution of this issue :slightly_smiling_face:

**Comment by YuriGal** [comment]
Hi, has this ever been addressed? We're still getting
```
Check: CKV_SECRET_3: "Azure Storage Account access key"
```
for AWS cloudformation template that has no Azure keys. This false positive forces us stuck on checkov v3.2.7 - the latest one before the issue was introduced, because any attempt to update, even to the latest version throws this false positive again.

**Comment by dvela-pan** [comment]
Got this false positive in Terraform planfiles for AWS architecture that has no Azure keys. It's generating massive amounts of noise and I can't go back to another version of Checkov because I'm a Prisma Cloud user.

**Comment by YuriGal** [comment]
I ended up adding `--skip-check CKV_SECRET_3` flag

**Comment by tsmithv11** [comment]
We've reversed some of the changes and improved this check. Can you try again? I tested on the code samples given here and they are not flagged anymore.

**Comment by rensvanleeuwen** [comment]
@tsmithv11

I can confirm with the latest version of Checkov, it no longer fails. The exclusion for `CKV_SECRET_3` has been removed, and the job runs successfully. 

Thank you!

**Comment by tsmithv11** [comment]
Glad to hear it! I'm closing this issue but please reopen it if you see a large volume of FPs for this check.
